### PR TITLE
CFG_CORE_SANITIZE_UNDEFINED

### PR DESCRIPTION
**Pull request by jenswi-linaro**

### CODE REVIEW

File: core/arch/arm/kernel/abort.c [hunk 1]
```diff
@@ -417,20 +417,21 @@ static bool is_abort_in_abort_handler(struct abort_info *ai __unused)
...
 
-#define A32_VTRANS64_MASK	A32_INSTR((0x7f << 21) | (7 << 9))
-#define A32_VTRANS64_VAL	A32_INSTR((0x62 << 21) | (5 << 9))
+#define A32_VTRANS64_MASK	A32_INSTR(SHIFT_U32(0x7f, 21) | SHIFT_U32(7, 9))
+#define A32_VTRANS64_VAL	A32_INSTR(SHIFT_U32(0x62, 21) | SHIFT_U32(5, 9))
 
-#define A32_VLDST_MASK		A32_INSTR((0xff  << 24) | (1 << 20))
-#define A32_VLDST_VAL		A32_INSTR((0xf4  << 24))
+#define A32_VLDST_MASK		A32_INSTR(SHIFT_U32(0xff, 24) | BIT32(20))
+#define A32_VLDST_VAL		A32_INSTR(SHIFT_U32(0xf4,24))
```
Comment (jbech-optee-adm) [inline core/arch/arm/kernel/abort.c]: Checkpatch is a bit grumpy over this particular line (missing space after comma).

Comment (jbech-linaro) [inline core/arch/arm/kernel/abort.c]: Checkpatch is a bit grumpy over this particular line (missing space after comma).

File: core/arch/arm/mm/core_mmu_lpae.c [hunk 2]
```diff
@@ -340,8 +340,9 @@ static struct tee_mmap_region *init_xlation_table(struct tee_mmap_region *mm,
 {
 	unsigned level_size_shift = L1_XLAT_ADDRESS_SHIFT - (level - 1) *
 						XLAT_TABLE_ENTRIES_SHIFT;
-	unsigned level_size = 1 << level_size_shift;
-	uint64_t level_index_mask = XLAT_TABLE_ENTRIES_MASK << level_size_shift;
+	unsigned level_size = BIT32(level_size_shift);
```
Comment (jbech-linaro) [inline core/arch/arm/mm/core_mmu_lpae.c]: Checkpatch also warns here:
`WARNING: Prefer 'unsigned int' to bare use of 'unsigned'`

**Comment by jbech-linaro** [comment]
With checkpatch fixes,
`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`

**Comment by jenswi-linaro** [comment]
Review comments addressed, but a few more bugs was discovered (and fixed) with gcc 5.3.

**Comment by jbech-linaro** [comment]
Thanks, I've also looked at the additional changes.
`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`

**Comment by jenswi-linaro** [comment]
Rebased and tags applied.

**Pull request merged**